File: AWS/Lambda/fromSNStoDynamoDB.js
var aws = require('aws-sdk');
var dynamodb = new aws.DynamoDB({params: {TableName: process.env.DYNAMODB_TABLE}});
var AWS_REGION = process.env.AWS_REGION;
var tableName = process.env.DYNAMODB_TABLE;


exports.handler = function(event, context) { 
    event = JSON.parse(event.Records[0].Sns.Message);
    var record = event.Records[0];
    var datetime = record.eventTime;
    var s3 = record.s3;
    var bucket = s3.bucket.name;
    var key = s3.object.key;
    var size = s3.object.size;
    var url = 'https://'+bucket+'.s3-'+AWS_REGION+'.amazonaws.com/'+key;
    
    dynamodb.putItem({
        TableName: tableName,
        Item : {
            filename: {S: key },
            eventTime: {S: datetime},
            url: {S: url },
            itemType: {S: 'image'}
        }
    }, function(err, data) {
        if (err) {
            console.error('error','putting item into dynamodb failed: '+err);
        }
        else {
            console.log('great success: '+JSON.stringify(data, null, '  '));
        }
    });

};



 

File: AWS/Lambda/fromSNStoDynamoDB.py
import json
import boto3
import os
import botocore

#Create a connection with DynamoDB Service
dynamodb = boto3.client('dynamodb')
#Get region of execution from environment variables available by default
AWS_REGION = os.environ['AWS_REGION']
#Get DynamoDB Table name from custom environment variable
# !! DO NOT HARD CODE THIS VALUE !!
var_tableName = os.environ['DYNAMODB_TABLE']

def lambda_handler(event, context):
    try:    
        #Parsing JSON from SNS event received
        message = json.loads(event['Records'][0]['Sns']['Message'])
    
        #Retrieving relevant information from the parsed message (Date, Bucket name, Object Name, Object Size)
        datetime = message['Records'][0]['eventTime']
        bucket = message['Records'][0]['s3']['bucket']['name']
        key = message['Records'][0]['s3']['object']['key']
        size = message['Records'][0]['s3']['object']['size']
    
        #Building the url to the object from retrived information
        url = 'https://' + bucket + 's3-' + AWS_REGION + ".amazonaws.com/" + key
    
        #Inserting item in DynamoDB
        response = dynamodb.put_item(
            Item={
                'filename': {'S': key },
                'eventTime': {'S': datetime},
                'url': {'S': url },
                'itemType': {'S': 'image'}
            },
            TableName = var_tableName,
        )
        
        if (response):
            print("Success! Below the result of what has been inserted in your DynamoDB table " + var_tableName)
            print(response)
            
    except botocore.exceptions.ClientError as e:
        print (e)
